include/event2/bufferevent_ssl.h

```diff
@@ -125,6 +125,81 @@ int bufferevent_ssl_renegotiate(struct bufferevent *bev);
 EVENT2_EXPORT_SYMBOL
 unsigned long bufferevent_get_openssl_error(struct bufferevent *bev);
 
+#endif
+#if defined(EVENT__HAVE_MBEDTLS) || defined(EVENT_IN_DOXYGEN_)
+struct mbedtls_ssl_context;
+/**
+   Create a new SSL bufferevent to send its data over another bufferevent.
+
+   @param base An event_base to use to detect reading and writing.  It
+      must also be the base for the underlying bufferevent.
+   @param underlying A socket to use for this SSL
+   @param ssl A SSL* object from openssl.
+   @param state The current state of the SSL connection
+   @param options One or more bufferevent_options
+   @return A new bufferevent on success, or NULL on failure
+*/
+EVENT2_EXPORT_SYMBOL
+struct bufferevent *
+bufferevent_mbedtls_filter_new(struct event_base *base,
+    struct bufferevent *underlying,
+    struct mbedtls_ssl_context *ssl,
+    enum bufferevent_ssl_state state,
+    int options);
+
+/**
+   Create a new SSL bufferevent to send its data over an SSL * on a socket.
+
+   @param base An event_base to use to detect reading and writing
+   @param fd A socket to use for this SSL
+   @param ssl A SSL* object from mbedtls.
+   @param state The current state of the SSL connection
+   @param options One or more bufferevent_options
+   @return A new bufferevent on success, or NULL on failure.
+*/
+EVENT2_EXPORT_SYMBOL
+struct bufferevent *
+bufferevent_mbedtls_socket_new(struct event_base *base,
```

Comment:
The main issue is that our regress_ssl does not run for the mbedtls, can you address this?

Reply:
I will add it later

Reply:
@okhowang  I'm prefer to address this before merging (since otherwise I don't know how this new wrapper works)

Reply:
sorry for reply later.
I'm busy on other work now, but I will done this recently

Reply:
I have add test for mbedtls.
it mainly based on test for openssl.